File: server_python/routers/blog_routes.py
from fastapi import APIRouter, Depends, File, Form, UploadFile
from middleware.auth import get_current_admin
from controllers.blog_controller import (
    add_blog,
    get_all_blogs,
    get_blog_by_id,
    delete_blog_by_id,
    toggle_publish,
    add_comment,
    get_blog_comments,
    get_topics,
    get_related_blogs,
    generate_content,
)

router = APIRouter()

@router.post("/add")
async def route_add(image: UploadFile | None = File(None), blog: str = Form(...), _: bool = Depends(get_current_admin)):
    return await add_blog(image, blog)

@router.get("/all")
async def route_all():
    return await get_all_blogs()

@router.get("/{blogId}")
async def route_by_id(blogId: str):
    return await get_blog_by_id(blogId)


@router.get("/related")
async def related_blogs(
    category: str,
    exclude: str
):
    return await get_related_blogs(category, exclude)

@router.post("/delete")
async def route_delete(payload: dict, _: bool = Depends(get_current_admin)):
    return await delete_blog_by_id(payload.get("id"))

@router.post("/toggle-publish")
async def route_toggle(payload: dict, _: bool = Depends(get_current_admin)):
    return await toggle_publish(payload.get("id"))

@router.post("/add-comment")
async def route_add_comment(payload: dict):
    return await add_comment(payload.get("blog"), payload.get("name"), payload.get("content"))

@router.post("/comments")
async def route_comments(payload: dict):
    return await get_blog_comments(payload.get("blogId"))

@router.post("/give-topics")
async def route_topics():
    return await get_topics()

@router.post("/generate")
async def route_generate(payload: dict, _: bool = Depends(get_current_admin)):
    return await generate_content(payload.get("prompt"))

File: server/routes/blogRoutes.js
import express from "express";
import {
  addBlog,
  addComment,
  deleteBlogById,
  generateContent,
  getAllBlogs,
  getBlogById,
  getBlogComments,
  togglePublish,
  getTopics,
  getRelatedBlogs,
  incrementViews,
  askAiAboutArticle,
  subscribeNewsletter,
  getBookmarkedBlogs,
  toggleBookmark
} from "../controllers/blogController.js";
import upload from "../middleware/multer.js";
import auth from "../middleware/auth.js";

const blogRouter = express.Router();

blogRouter.post("/add", upload.single('image'), auth, addBlog);
blogRouter.get('/all', getAllBlogs);
blogRouter.get('/related', getRelatedBlogs);
blogRouter.post('/subscribe', subscribeNewsletter);
blogRouter.post('/:blogId/view', incrementViews);
blogRouter.post('/:blogId/ask-ai', askAiAboutArticle);
blogRouter.get("/bookmarks",auth,getBookmarkedBlogs);
blogRouter.get('/:blogId', getBlogById);
blogRouter.post('/delete', auth, deleteBlogById);
blogRouter.post('/toggle-publish', auth, togglePublish);
blogRouter.post('/add-comment', addComment);
blogRouter.post('/comments', getBlogComments);
blogRouter.post('/give-topics', auth, getTopics);
blogRouter.post('/generate', auth, generateContent);
blogRouter.post("/bookmark",auth,toggleBookmark);

export default blogRouter;